### PR TITLE
Case 20060: Grab correct stencil masks from display plugins

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20060/We-are-artificially-clipping-Oculus-FOV

Also updated LibOVR to 1.35

Test plan (test on Rift, Vive, and desktop):
- Go into HMD. You shouldn't see extra black borders around your field of view.  In the desktop mono preview, you shouldn't see the black borders.  In the desktop stereo preview, you SHOULD see black borders around the two eye views.
- Mirrors should appear normal in desktop and HMD.
- You should be able to take still, GIF, and 360 snapshots in HMD and desktop.  To take a 360 snapshot, you can enter this in the console (you might need to go into 3rd person in HMD):
`Window.takeSecondaryCamera360Snapshot(Camera.position)`

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18777/HighFidelity-Beta-PR15405-e7d12dc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18777/HighFidelity-Beta-Interface-PR15405-e7d12dc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18777/HighFidelity-Beta-PR15405-e7d12dc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18777/HighFidelity-Beta-Interface-PR15405-e7d12dc.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18777/HighFidelity-Beta-PR15405-e7d12dc.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18777/HighFidelity-Quest-Beta-PR15405-e7d12dc.apk)

**Comment by JeffClinton** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18871/HighFidelity-Beta-PR15405-e7d12dc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18871/HighFidelity-Beta-Interface-PR15405-e7d12dc.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18871/HighFidelity-Beta-PR15405-e7d12dc.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18871/HighFidelity-Quest-Beta-PR15405-e7d12dc.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18871/HighFidelity-Beta-PR15405-e7d12dc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18871/HighFidelity-Beta-Interface-PR15405-e7d12dc.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18891/HighFidelity-Beta-PR15405-4fc97ef.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18891/HighFidelity-Quest-Beta-PR15405-4fc97ef.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18892/HighFidelity-Beta-PR15405-018d249.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18892/HighFidelity-Quest-Beta-PR15405-018d249.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18892/HighFidelity-Beta-PR15405-018d249.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18892/HighFidelity-Beta-Interface-PR15405-018d249.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18910/HighFidelity-Beta-PR15405-5726503.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18910/HighFidelity-Beta-Interface-PR15405-5726503.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18910/HighFidelity-Beta-PR15405-5726503.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18910/HighFidelity-Quest-Beta-PR15405-5726503.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18913/HighFidelity-Beta-PR15405-ab394e6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18913/HighFidelity-Beta-Interface-PR15405-ab394e6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18913/HighFidelity-Beta-PR15405-ab394e6.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18913/HighFidelity-Quest-Beta-PR15405-ab394e6.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18919/HighFidelity-Beta-PR15405-48aef3e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18919/HighFidelity-Beta-Interface-PR15405-48aef3e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18919/HighFidelity-Beta-PR15405-48aef3e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18919/HighFidelity-Quest-Beta-PR15405-48aef3e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18919/HighFidelity-Beta-PR15405-48aef3e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18919/HighFidelity-Beta-Interface-PR15405-48aef3e.exe

File: interface/src/ui/SnapshotAnimated.cpp [hunk 1]
```diff
@@ -62,44 +60,45 @@ void SnapshotAnimated::saveSnapshotAnimated(QString pathStill, float aspectRatio
...
+        // This also pops up the "Share" dialog. The unprocessed GIF will be visualized as a loading icon until processingGifCompleted() is called.
+        emit SnapshotAnimated::snapshotAnimatedDM->processingGifStarted(SnapshotAnimated::snapshotStillPath);
+
+        // Kick off the thread that'll pack the frames into the GIF
+        QtConcurrent::run(processFrames);
+        // Stop the snapshot QTimer. This action by itself DOES NOT GUARANTEE
+        // that the slot will not be called again in the future.
+        // See: http://lists.qt-project.org/pipermail/qt-interest-old/2009-October/013926.html
+        SnapshotAnimated::snapshotAnimatedTimer->stop();
+        delete SnapshotAnimated::snapshotAnimatedTimer;
```
Comment (AndrewMeadows) [inline interface/src/ui/SnapshotAnimated.cpp]: I was told it is ok to `disconnect` the `slot` from its `signal` inside the `slot` itself.  Perhaps you want to do that before the `delete`?

Comment (SamGondelman) [reply interface/src/ui/SnapshotAnimated.cpp]: I think we could do a separate pass to clean up the code in these snapshot classes.  There's a number of things that could be improved.

Comment (AndrewMeadows) [reply interface/src/ui/SnapshotAnimated.cpp]: I was suggesting it to address the "no guarantee the slot will not be called" problem the comment was talking about.

Comment (SamGondelman) [reply interface/src/ui/SnapshotAnimated.cpp]: Yeah, I didn't add that, I just moved the code from before around, I was just agreeing that improvements could be made here and elsewhere and we could do it all at once

**Pull request merged**